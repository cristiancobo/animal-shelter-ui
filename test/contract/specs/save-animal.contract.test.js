
import {init_provider} from '../config/initPact';
import { AnimalController } from "../../../controllers";
import {Matchers} from "@pact-foundation/pact";

const animal = {
    name: "Loli",
    breed: "Birmano",
    gender: "Male",
    vaccinated:true,
}


const provider = init_provider();
describe('Given An Animal Service', () =>{ 
    beforeAll(async ()=>{
        await provider.setup();
    });

   describe('When a request the animals create', () =>{
        beforeAll(async ()=>{
            await provider.addInteraction({
                uponReceiving: 'a request to get the animal created',
                state:"save animal",
                withRequest: {
                    method: 'POST',
                    path: '/animals',	
                    body: animal,		
                },
                willRespondWith: {
                    status:201,
                    body: Matchers.like(animal),
                }
            });
       });
        it("Then it should return the right data", async() => {
            const response = await AnimalController.register(animal);
            expect(response.data).toMatchSnapshot();        
            await provider.verify();
        });
    });
    afterAll(async ()=>{
        await provider.finalize();
    });
});