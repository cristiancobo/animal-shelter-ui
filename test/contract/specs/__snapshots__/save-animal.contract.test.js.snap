// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given An Animal Service When a request the animals create Then it should return the right data 1`] = `
Object {
  "breed": "Birmano",
  "gender": "Male",
  "name": "Loli",
  "vaccinated": true,
}
`;
